src/main/java/com/marklogic/spark/writer/MarkLogicWrite.java

```diff
@@ -53,4 +58,25 @@ public void abort(WriterCommitMessage[] messages) {
             logger.error("Abort messages received: {}", Arrays.asList(messages));
         }
     }
+
+    @Override
+    public StreamingDataWriterFactory createStreamingWriterFactory(PhysicalWriteInfo info) {
+        return new MarkLogicDataWriterFactory(writeContext);
+    }
+
+    @Override
+    public void commit(long epochId, WriterCommitMessage[] messages) {
+        logger.info("EpochId for commit is "+ epochId);
```

Comment:
Small thing, which I'll tweak later - `logger` functions allow for token replacement, so you can do `logger.info("epochId for commit is {}", epochId)` . I keep trying to remember that, I know Sonar (which we don't yet have running) will complain about it otherwise.